app/src/main/java/com/rentals/apartment/controller/ApartmentsApi.java

```diff
@@ -20,17 +22,10 @@ public ApartmentsApi(ApartmentService apartmentService) {
 
     @GetMapping("/apartments")
     public List<ApartmentRecord> getAllApartments(
-            @RequestParam(required = false) String bedrooms,
-            @RequestParam(required = false) String bathrooms,
-            @RequestParam(required = false) String minArea,
-            @RequestParam(required = false) String maxArea,
-            @RequestParam(required = false) boolean hasParking,
-            @RequestParam(required = false) String minPrice,
-            @RequestParam(required = false) String maxPrice,
-            @RequestParam(required = false) String description,
+            @RequestParam Map<String, String> filter,
```

Comment:
if we remove @RequestParam and just lose the map, changing to a filter object, we can make this much simpler